include/irr/core/math/aabb2dSIMDIntBase.h

```diff
@@ -216,9 +175,80 @@ class aabbSIMD<core::vector2df_SIMD> : public aabbSIMDBase<aabbSIMD, core::vecto
 
 };
 
+template<typename VECTOR_TYPE>
+class aabb2dSIMDInt : public aabbSIMDBase<aabb2dSIMDInt, VECTOR_TYPE>
+{
+	static_assert(
+		std::is_same<VECTOR_TYPE, core::vector2di32_SIMD>::value ||
+		std::is_same<VECTOR_TYPE, core::vector2du32_SIMD>::value,
+		"assertion failed: cannot use this type");
+
+public:
+	inline aabb2dSIMDInt()
+	{
+		if constexpr (std::is_same<VECTOR_TYPE, core::vector2di32_SIMD>::value)
+			internalBoxRepresentation = core::vector4di32_SIMD(INT_MIN);
+
+		if constexpr (std::is_same<VECTOR_TYPE, core::vector2du32_SIMD>::value)
```

Comment:
We don't quite support C++17, but we have out own workaround called `irr::static_if` if I remember correctly

Reply:
whats the reason to not support c++17?

Reply:
having to compile on old versions of GCC for edu.buildaworld.net
(we support c++17, but need fallbacks, `static_if` is a fallback for c++14)